sdks/go/pkg/beam/runners/universal/runnerlib/job.go

```diff
@@ -67,16 +67,16 @@ func Prepare(ctx context.Context, client jobpb.JobServiceClient, p *pb.Pipeline,
 	}
 	resp, err := client.Prepare(ctx, req)
 	if err != nil {
-		return "", "", fmt.Errorf("failed to connect to job service: %v", err)
+		return "", "", "", fmt.Errorf("failed to connect to job service: %v", err)
 	}
-	return resp.GetPreparationId(), resp.GetArtifactStagingEndpoint().GetUrl(), nil
+	return resp.GetPreparationId(), resp.GetArtifactStagingEndpoint().GetUrl(), resp.GetStagingSessionToken(), nil
 }
 
 // Submit submits a job to the given job service. It returns a jobID, if successful.
 func Submit(ctx context.Context, client jobpb.JobServiceClient, id, token string) (string, error) {
 	req := &jobpb.RunJobRequest{
-		PreparationId: id,
-		RetrievalToken:  token,
+		PreparationId:  id,
```

Comment:
Hmm where does the "token" here come from? It ought to be the retrieval token, obtained from CommitManifest. I'm surprised that this argument even existed here, before the PR that introduced the retrieval token.

Reply:
Uncertain, this code only changed  due to formatting for whatever reason. 
Prior to your CL a token was returned from the commit requests, so it's presumably that.